### DISCUSSION TITLE
instal error for Pmetric in Rstudio

### DISCUSSION THREAD

**Discussion by fabianyskong** [LAPKB/Pmetrics#193] [General]
Hello.
I am trying to instal Pmetric to Rstudio (details below) 
- RStudio 2023.12.0+369 "Ocean Storm" Release (33206f75bd14d07d84753f965eaa24756eda97b7, 2023-12-17) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.12.0+369 Chrome/116.0.5845.190 Electron/26.2.4 Safari/537.36

I am tryping > devtools::install_github("LAPKB/Pmetrics") then get below error message

Warning message:
In i.p(...) :
  installation of package ‘C:/Users/kongf/AppData/Local/Temp/RtmpuafXCK/file12e01ef8fcf/Pmetrics_2.1.1.tar.gz’ had non-zero exit status

how do I resolve this? I am new to Rstudio and Pmetric (coding novice overall)
Thanks

**Comment by Siel** [comment]
Hi @fabianyskong,
I would love to help you, I might need the whole error log, since what I'm seeing (the warning) does not provide enough information.

Thanks!

**Comment by fabianyskong** [comment]
Hi Julian
Thanks so much for the help. Below were my steps I took using the instructions from http://www.lapk.org/Pmetrics_install.php


  *   already have R and Rstudio installed.
  *   I installed rtools 4.2
  *   followed instructions from http://www.lapk.org/Pmetrics_install.php

     *
Installed "devtools" [successfully]
     *   then this step failed devtools::install_github("LAPKB/Pmetrics")
the code below


> devtools::install_github("LAPKB/Pmetrics")
Downloading GitHub repo ***@***.***
── R CMD build ────────────────────────────────────────────────────────────────────────────────────────✔  checking for file 'C:\Users\kongf\AppData\Local\Temp\RtmpMTNfBO\remotes5cf06a2a3cbc\LAPKB-Pmetrics-ae7d0d3/DESCRIPTION' (1.7s)─  preparing 'Pmetrics': (15.7s)✔  checking DESCRIPTION meta-information ...
─  cleaning src─  checking for LF line-endings in source and make files and shell scripts (4.1s)─  checking for empty or unneeded directories
   Removed empty directory 'Pmetrics/inst/Examples/Runs'
─  building 'Pmetrics_2.1.1.tar.gz'

Installing package into ‘C:/Users/kongf/AppData/Local/R/win-library/4.2’
(as ‘lib’ is unspecified)
* installing *source* package 'Pmetrics' ...
** using staged installation
** libs
g++ -std=gnu++11  -I"C:/PROGRA~1/R/R-42~1.1/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c knn.cpp -o knn.o
g++ -std=gnu++11  -I"C:/PROGRA~1/R/R-42~1.1/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c registerDynamicSymbol.cpp -o registerDynamicSymbol.o
g++ -std=gnu++11 -shared -s -static-libgcc -o Pmetrics.dll tmp.def knn.o registerDynamicSymbol.o -LC:/rtools42/x86_64-w64-mingw32.static.posix/lib/x64 -LC:/rtools42/x86_64-w64-mingw32.static.posix/lib -LC:/PROGRA~1/R/R-42~1.1/bin/x64 -lR
installing to C:/Users/kongf/AppData/Local/R/win-library/4.2/00LOCK-Pmetrics/00new/Pmetrics/libs/x64
** R
** data
*** moving datasets to lazyload DB
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'ITex'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'ITex'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'NPex'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'NPex'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'NPex_val'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'NPex_val'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'badData'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'badData'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'dataEx'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'dataEx'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'modEx'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'modEx'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'modelLibrary'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'modelLibrary'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'simEx'
Warning: namespace 'Pmetrics' is not available and has been replaced
by .GlobalEnv when processing object 'simEx'
** inst
** byte-compile and prepare package for lazy loading
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) :
  there is no package called 'mvtnorm'
Calls: <Anonymous> ... loadNamespace -> withRestarts -> withOneRestart -> doWithOneRestart
Execution halted
ERROR: lazy loading failed for package 'Pmetrics'
* removing 'C:/Users/kongf/AppData/Local/R/win-library/4.2/Pmetrics'
Warning message:
In i.p(...) :
  installation of package ‘C:/Users/kongf/AppData/Local/Temp/RtmpMTNfBO/file5cf01aca13b3/Pmetrics_2.1.1.tar.gz’ had non-zero exit status

Let me know what you think. This happened both with my desktop at work (with the Windows details I posted - Windows 10)

I have the same issue with my laptop at home - laptop details are below

RStudio 2023.12.0+369 "Ocean Storm" Release (33206f75bd14d07d84753f965eaa24756eda97b7, 2023-12-17) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.12.0+369 Chrome/116.0.5845.190 Electron/26.2.4 Safari/537.36

Thanks alot!


Associate Professor Fabian Kong BPharm, MEpi, PhD (He/Him/His)

Deputy Head | C.R. Roper Fellow



Sexual Health Unit - Centre for Epidemiology and Biostatistics

Melbourne School of Population and Global Health, The University of Melbourne, Australia.

Room 351, Level 3, 207 Bouverie St, Parkville, 3053, Australia.

e: ***@***.******@***.***> | tel: +613 9035 3039 | mob: +61 423 115 203

 [cid:ff746d9e-9c8f-4b15-88f0-2b871bf73e44] @fabian_kong<https://twitter.com/fabian_kong>


 [cid:ea8a3167-9c5d-4010-a6ba-58bb9ca3841c]






________________________________

**Comment by Siel** [comment]
Hi Fabian,

From what I can see the specific error that is causing your install to fail
is that the package you are missing the mvtnorm package, you can install it
using R studio, or by typing install.packages("mvtnorm") in your R terminal.

Please let me know how it goes! and thanks for using Pmetrics :)
*__________________________*
Electronic Engineer
PhD in Electronic Engineering and Computer Sciences.
Researcher at LAPKB - CHLA/USC

**Comment by fabianyskong** [comment]
Hi Juilan
That worked! thanks. Below FYI



> install.packages("mvtnorm")
Installing package into ‘C:/Users/kongf/AppData/Local/R/win-library/4.2’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.2/mvtnorm_1.2-4.zip'
Content type 'application/zip' length 753051 bytes (735 KB)
downloaded 735 KB

package ‘mvtnorm’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
        C:\Users\kongf\AppData\Local\Temp\RtmpCCtqhr\downloaded_packages
> devtools::install_github("LAPKB/Pmetrics")
Downloading GitHub repo ***@***.***
── R CMD build ────────────────────────────────────────────────────────────────────────────────────────✔  checking for file 'C:\Users\kongf\AppData\Local\Temp\RtmpCCtqhr\remotes3b406be8c4a\LAPKB-Pmetrics-ae7d0d3/DESCRIPTION' (759ms)─  preparing 'Pmetrics': (6.8s)✔  checking DESCRIPTION meta-information ...
─  cleaning src─  checking for LF line-endings in source and make files and shell scripts (1.7s)─  checking for empty or unneeded directories
   Removed empty directory 'Pmetrics/inst/Examples/Runs'
─  building 'Pmetrics_2.1.1.tar.gz'

Installing package into ‘C:/Users/kongf/AppData/Local/R/win-library/4.2’
(as ‘lib’ is unspecified)
* installing *source* package 'Pmetrics' ...
** using staged installation
** libs
g++ -std=gnu++11  -I"C:/PROGRA~1/R/R-42~1.1/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c knn.cpp -o knn.o
g++ -std=gnu++11  -I"C:/PROGRA~1/R/R-42~1.1/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c registerDynamicSymbol.cpp -o registerDynamicSymbol.o
g++ -std=gnu++11 -shared -s -static-libgcc -o Pmetrics.dll tmp.def knn.o registerDynamicSymbol.o -LC:/rtools42/x86_64-w64-mingw32.static.posix/lib/x64 -LC:/rtools42/x86_64-w64-mingw32.static.posix/lib -LC:/PROGRA~1/R/R-42~1.1/bin/x64 -lR
installing to C:/Users/kongf/AppData/Local/R/win-library/4.2/00LOCK-Pmetrics/00new/Pmetrics/libs/x64
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
Language has changed. Based on system, setting default language to .
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (Pmetrics)
> library(Pmetrics)
Welcome to Pmetrics, version 2.1.1.
Use PMmanual() or visit the LAPK website at http://www.lapk.org/pmetrics.php for help.
Use PM_tutorial() for an introduction.
The Pmetrics forum is back! Click the tab at www.lapk.org for questions, tips and bug reports.

Follow Pmetrics tweets at https://twitter.com/USCLAPK.


CRITICAL: Execute PMbuild() in R to complete Pmetrics installation.

> PMbuild()
Pmetrics cannot find required compiled binary files.
C:\rtools42\x86_64-w64-mingw32.static.posix\bin\gfortran.exe
Pmetrics has detected gfortran and will compile required binary files.
If the registration time outs, please re-run PMbuild(skipRegistration = TRUE) Pmetrics has anonymously registered your installation of this version.
LAPKB does not collect or store any personal or identifying information.

Compiling 1 of 9: NPprep...
Compiling 2 of 9: NPeng...
Compiling 3 of 9: ITprep...
Compiling 4 of 9: ITeng...
Compiling 5 of 9: ITerr...
Compiling 6 of 9: SIMeng...
Compiling 7 of 9: DOprep...
Compiling 8 of 9: DOeng...
Compiling 9 of 9: mb2csv...
All packages installed and permanent Fortran modules compiled.
> PMtest()
You have the Fortran source files

________________________________

**Comment by fabianyskong** [comment]
Hi Julian
I have registered for the 3 day intensive in March with Jason Roberts
I'm looking at the program so this is why I have installed Pmetric
I see you will also use BestDose but this is by request. Can I grab this and any installation instructions?
Thanks
________________________________

**Comment by Siel** [comment]
Hi @fabianyskong, that is great, I'll see you there! Regarding best dose, I might need to ask professor Neely how are we planning to distribute the installer for the workshop. I'm sure you will receive more information in the next couple of weeks. For the time being I'll mark this issue as closed, but feel free to re-open it if you still need some help.

Julian

**Comment by fabianyskong** [comment]
Hi Julian
Ok great. Thanks for the help.
________________________________
